core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -173,21 +194,30 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
     info("Stopping socket server request processors")
     this.synchronized {
       acceptors.asScala.values.foreach(_.shutdown())
-      processors.asScala.values.foreach(_.shutdown())
-      requestChannel.clear()
+      dataProcessors.asScala.values.foreach(_.shutdown())
+      controlPlaneProcessors.asScala.values.foreach(_.shutdown())
```

Comment:
with the above suggestion, this would be : 
```suggestion
     controlPlaneProcessorOpt.map(_.shutdown())
```